public/app/features/transformers/configFromQuery/configFromQuery.ts

```diff
@@ -73,6 +74,27 @@ export function extractConfigFromQuery(options: ConfigFromQueryTransformOptions,
     for (const field of frame.fields) {
       if (matcher(field, frame, data)) {
         const dataConfig = getFieldConfigFromFrame(reducedConfigFrame, 0, mappingResult);
+
+        if (options.isDisplayNameMapping) {
```

Comment:
@leeoniya when you get a chance, do you have any thoughts on where this should go? I was thinking I could use some of the logic we already have, but found this is too different than what we have already. I thought too about using this as a way to build a field override to do what we want, but the field overrides for display name might be bulky to construct if we make one override for every key/value rename. What do you think?